### PR TITLE
[CDAP-6204] Added configurable number of features to NaiveBayesTrainer and NaiveBayesClassifier

### PR DESCRIPTION
**Pull request by knazir**
Changed the number of features used by both NaiveBayesTrainer and NaiveBayesClassifier from a hardcoded 100 to a configurable parameter.

### CODE REVIEW

File: spark-plugins/src/main/java/co/cask/hydrator/plugin/batch/spark/NaiveBayesClassifier.java [hunk 1]
```diff
@@ -76,11 +76,25 @@
...
       this.fileSetName = fileSetName;
       this.path = path;
       this.fieldToClassify = fieldToClassify;
       this.predictionField = predictionField;
+      this.numFeatures = numFeatures;
+    }
+
+    public boolean validateNumFeatures() {
+      try {
+        int parsedNumFeatures = Integer.parseInt(numFeatures.trim());
```
Comment (anwar6953) [inline spark-plugins/src/main/java/co/cask/hydrator/plugin/batch/spark/NaiveBayesClassifier.java]: NumberFormatException is a subclass of IllegalArgumentException, so should be fine to simply allow that to propagate that (don’t catch and rethrow something else)

**Comment by knazir** [comment]
Build and unit tests: http://builds.cask.co/browse/HYP-DBT14-2/issues

File: spark-plugins/docs/NaiveBayesClassifier-sparkcompute.md [hunk 2]
```diff
@@ -19,6 +19,9 @@ Properties
 
 **predictionField:** The field on which to set the prediction. It will be of type double.
 
+**numFeatures:** The number of features to use when classifying with the trained model. This should be the same as
+the number of features used to train the model in NaiveBayesTrainer.
```
Comment (sreevatsanraman) [inline spark-plugins/docs/NaiveBayesClassifier-sparkcompute.md]: Specify the default value that will be used.

File: spark-plugins/docs/NaiveBayesTrainer-sparksink.md [hunk 3]
```diff
@@ -21,6 +21,9 @@ Properties
 
 **predictionField:** The field from which to get the prediction. It must be of type double.
 
+**numFeatures:** The number of features to train the model with. This should be the same as the number of features
+used for the NaiveBayesClassifier.
```
Comment (sreevatsanraman) [inline spark-plugins/docs/NaiveBayesTrainer-sparksink.md]: Same here. Specify default

File: spark-plugins/src/main/java/co/cask/hydrator/plugin/batch/spark/NaiveBayesClassifier.java [hunk 4]
```diff
@@ -76,11 +76,17 @@
     @Description("The field on which to set the prediction. It will be of type double.")
     private final String predictionField;
 
-    public Config(String fileSetName, String path, String fieldToClassify, String predictionField) {
+    @Description("The number of features to use in training the model. It must be of type integer and equal to the " +
```
Comment (sreevatsanraman) [inline spark-plugins/src/main/java/co/cask/hydrator/plugin/batch/spark/NaiveBayesClassifier.java]: same here. default.

File: spark-plugins/src/main/java/co/cask/hydrator/plugin/batch/spark/NaiveBayesTrainer.java [hunk 5]
```diff
@@ -66,11 +66,17 @@
     @Description("The field from which to get the prediction. It must be of type double.")
     private final String predictionField;
 
-    public Config(String fileSetName, String path, String fieldToClassify, String predictionField) {
+    @Description("The number of features to use in training the model. It must be of type integer and equal to the" +
```
Comment (sreevatsanraman) [inline spark-plugins/src/main/java/co/cask/hydrator/plugin/batch/spark/NaiveBayesTrainer.java:71]: default.

**Comment by sreevatsanraman** [comment]
Minor comments. Fix and merge.

**Comment by knazir** [comment]
Tests looks good: http://builds.cask.co/browse/HYP-DBT14-10 going to merge.

**Pull request merged**